### PR TITLE
No Slotting Arithmetic

### PR DESCRIPTION
**Pull request by KtorZ**
# Issue Number

#94

# Overview

- [ ] I have remove slotting arithmetic operation from the source code (relying instead on the fact that blocks are linked to each other and points to their parent)
- [ ] I have added a few tests to make sure that ordering of slots happens as expect, in a sudden paranoia crisis (since we derive the `Ord` automatically, it _just happened_ that we defined the epoch index before the slot number in the record declaration. Swapping those two would have dramatic effects ^^")

# Comments

As a side-effect, we don't have to test for any of the boundary cases because there are none :tada:

### CODE REVIEW

File: test/unit/Cardano/Wallet/Primitive/TypesSpec.hs [hunk 1]
```diff
@@ -339,15 +291,10 @@ instance Arbitrary BlockHeader where
...
-newtype NumberOfSlots = NumberOfSlots Word64 deriving Show
-
-instance Arbitrary NumberOfSlots where
-    shrink (NumberOfSlots 0) = []
-    shrink (NumberOfSlots n) = NumberOfSlots <$> [0, n `div` 2, n - 1]
-    arbitrary = NumberOfSlots . fromIntegral
-        <$> choose (0 :: Int, 4 * (fromIntegral slotsPerEpoch))
+    arbitrary = do
+        ep <- choose (0, 10)
+        sl <- choose (0, 100)
```
Comment (piotr-iohk) [inline test/unit/Cardano/Wallet/Primitive/TypesSpec.hs:296]: why not `sl <- choose (0, 21600)`?
Also is it possible to have slot 0? or epoch 0?
It is because it's just an Id I suppose?

Comment (piotr-iohk) [reply test/unit/Cardano/Wallet/Primitive/TypesSpec.hs:296]: hm, is it even needed? I mean the whole generator for `SlotId`? The [`Slotting ordering`](https://github.com/input-output-hk/cardano-wallet/pull/126/files#diff-00fd2cdc39324db62230aba60bf1287bR87) test suite ^ is using only some explicit values for SlotId...

Comment (KtorZ) [reply test/unit/Cardano/Wallet/Primitive/TypesSpec.hs:296]: Not only, there's a first property that compare arbitrary slots against the initial `SlotId 0 0` (making sure they're all greater than it).

Why not `21600` ? Well, to really show that it doesn't matter here and is completely irrelevant for the testing of this module. The only thing we really test is that we compare slot correctly. There's no other assumption about `SlotId` (and that's what we try to preserve in the code as well).

**Review by piotr-iohk — APPROVED** [review]
:+1:

**Comment by KtorZ** [comment]
Merged into #128 to serialize PR / CI builds
